src/davies_meyer.rs

```diff
@@ -11,7 +11,7 @@ use crate::gadget_traits::{BlockCipher, CompressionFunction};
 /// cipher.
 pub struct DaviesMeyer<F: Field, BC: BlockCipher<F>> {
     cipher: BC,
-    phantom: PhantomData<F>,
+    phantom: PhantomData<*const F>,
```

Comment:
`PhantomData<*const F>` is the preferred declaration according to the documentation.